### ISSUE TITLE
Update SimpleScheduler to support Conditions

### ISSUE THREAD

**Issue by pgleeson** [ModECI/MDF#30]
The [SimpleScheduler](https://github.com/ModECI/MDF/blob/main/modeci_mdf/SimpleScheduler.py), which can execute simple MDF models does not currently support conditions: https://github.com/ModECI/MDF#conditions.

The [definitions of the python MDF](https://github.com/ModECI/MDF/blob/main/modeci_mdf/MDF.py) should be updated to support conditions and a basic interpretation/handling of these should be added to the scheduler.

**Comment by kmantel** [comment]
I've added a very basic definition for conditions in #37. 

Regarding the scheduler - I'm assuming this would be a porting of the psyneulink scheduler and partially conditions. A lot of the scheduler and condition logic is generic, only requiring a DAG as input. However, there are a few things assumed to be as they are in psyneulink that would complicate the port. First is "time scales": 

- RUN: a user-initiated call consisting of a batch of inputs to the model
- TRIAL: all activity that occurs during the processing of a single input 
- PASS: the time within a TRIAL in which all nodes in a graph have had the chance to execute at least once (when a node has a chance to execute, this execution may still be blocked by its run condition)
- TIME_STEP: the smallest unit of time which consists of any nodes that actually execute and do so simultaneously

While there was some discussion on opening up these to definition by an environment, and I've been able to come up with a sort of informal language to describe them in the mdf, an obvious way to remove these assumptions from the scheduler and actually implement environment-defined time scales isn't clear to me.

Second, absolute/real time is currently not implemented in the scheduler. There was some work done that would eventually be able to support this, but I haven't had a chance to fully understand the changes and whether they make any more psyneulink-specific assumptions. I can say that more still needs to be done to support absolute time at all.
